packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,260 @@
+  }
+
+  public logRequest(request: HttpRequest): void {
+    if (!this._isConfigured) {
+      return;
+    }
+
+    const logLevel = this._loggingOptions.logLevel ?? Level.Info;
+    const contentTypeHeader = this.getContentType(request.headers);
+    const url = this._loggingOptions.logRequest?.includeQueryInPath
+      ? request.url
+      : this.removeQueryParams(request.url);
+
+    this._logger.log(
+      logLevel,
+      'Request  HttpMethod: ${method} Url: ${url} ContentType: ${contentType}',
+      {
+        method: request.method,
+        url,
+        contentType: contentTypeHeader,
+      }
+    );
+
+    this.applyLogRequestOptions(logLevel, request);
+  }
+
+  public logResponse(response: HttpResponse): void {
+    if (!this._isConfigured) {
+      return;
+    }
+
+    const logLevel = this._loggingOptions.logLevel ?? Level.Info;
+    const contentTypeHeader = this.getContentType(response.headers);
+    const contentLengthHeader = this.getContentLength(response.headers);
+
+    this._logger.log(
+      logLevel,
+      'Response HttpStatusCode: ${statusCode} Length: ${contentLength} ContentType: ${contentType}',
+      {
+        statusCode: response.statusCode,
+        contentLength: contentLengthHeader,
+        contentType: contentTypeHeader,
+      }
+    );
+
+    this.applyLogResponseOptions(logLevel, response);
+  }
+
+  private applyLogRequestOptions(level: Level, request: HttpRequest) {
+    this.applyLogRequestHeaders(
+      level,
+      request,
+      this._loggingOptions.logRequest
+    );
+
+    this.applyLogRequestBody(level, request, this._loggingOptions.logRequest);
+  }
+
+  private applyLogRequestHeaders(
+    level: Level,
+    request: HttpRequest,
+    logRequest?: HttpRequestLoggingOptions
+  ) {
+    if (logRequest) {
+      const { logHeaders, headerToInclude, headerToExclude } = logRequest;
+
+      if (logHeaders) {
+        const headersToLog = this.extractHeadersToLog(
+          request.headers,
+          headerToInclude,
+          headerToExclude
+        );
+
+        this._loggingOptions.logger?.log(level, 'Request Headers: ${headers}', {
+          headers: headersToLog,
+        });
+      }
+    }
+  }
+
+  private applyLogRequestBody(
+    level: Level,
+    request: HttpRequest,
+    logRequest?: HttpRequestLoggingOptions
+  ) {
+    if (logRequest?.logBody) {
+      this._loggingOptions.logger?.log(level, 'Request Body: ${body}', {
+        body: request.body,
+      });
+    }
+  }
+
+  private applyLogResponseOptions(level: Level, response: HttpResponse) {
+    this.applyLogResponseHeaders(
+      level,
+      response,
+      this._loggingOptions.logResponse
+    );
+
+    this.applyLogResponseBody(
+      level,
+      response,
+      this._loggingOptions.logResponse
+    );
+  }
+
+  private applyLogResponseHeaders(
+    level: Level,
+    response: HttpResponse,
+    logResponse?: HttpMessageLoggingOptions
```

Comment:
There won't be any need to have this parameter as optional once you fix the issue I mentioned in the logging options interfaces.